pkgs/development/libraries/ffmpeg/generic.nix

```diff
@@ -826,8 +827,7 @@ stdenv.mkDerivation (
       perl
       pkg-config
     ]
-    # 8.0 is only compatible with nasm, and we don't want to rebuild all older ffmpeg builds at this moment.
-    ++ (if versionOlder version "8.0" then [ yasm ] else [ nasm ])
+    ++ optionals stdenv.hostPlatform.isx86 [ nasm ]
```

Comment:
This is semantically different. Previously nasm was used on any platform as long as the version was greater than 8.0. Now it only uses nasm on x86. `nix eval name.meta.platforms` indicates that it works on many more platforms beyond x86, so why change it?

Reply:
and ffmpeg it self also only ever made use of yasm and nasm when on x86 and otherwise just ignored it.